### ISSUE TITLE
Termination checker runs too early for definitions inside record (or: positivity checker runs too late)

### ISSUE THREAD

**Issue by vjackson725** [agda/agda#2975]
```Agda
record Foo {x} (X : Set x) : Set x where
  coinductive
  field
    force : Foo X

-- module _ {x} (X : Set x) where
--   open Foo

  forever : X → Foo X
  force (forever x) = forever x
```

```
An internal error has occurred. Please report this as a bug.
Location of the error: src/full/Agda/TypeChecking/Monad/Base.hs:1664
```

This does not occur when the same definition is in a module.

**Comment by nad** [comment]
Here is a slightly smaller test case:
```agda
record R : Set where
  coinductive
  field force : R

  r : R
  force r = r
```
This seems to be a regression introduced with the release of Agda 2.5.3.

**Comment by andreasabel** [comment]
The problem seems to be that the positivity checker for `R` has not run when the termination checking for `r` is attempted.  There, Agda wants to see whether `force` is a recursive projection, but recursivity of `R` has not been computed yet, thus, Agda crashes.

**Comment by andreasabel** [comment]
> This seems to be a regression introduced with the release of Agda 2.5.3.

Well, maybe the problem was amplified by the refactoring e7240e844e0d670eb471f983fd26922cdb4af3d0 .  However, Agda 2.5.2 reports a termination error, for the same reason that Agda 2.5.3 crashes: It cannot yet see that `force` is a projection from a recursive coinductive type.  Thus, both releases reject the test case, and are wrong in doing so.

**Comment by andreasabel** [comment]
I am replacing the internal error by a termination error.  A proper fix could be done together with #434 / #2500.

**Comment by andreasabel** [comment]
My fix for #2963 seems to also fix this issue.

**Comment by andreasabel** [comment]
"regression in master" is reserved for yet unreleased regressions in the development version.
